packages/nestjs/src/setup.ts

```diff
@@ -0,0 +1,153 @@
+        getIsolationScope().setTransactionName(`${req.method?.toUpperCase() || 'GET'} ${req.route.path}`);
+      }
+    }
+
+    return next.handle();
+  }
+}
+
+/**
+ *
+ */
+export class SentryGlobalFilter extends BaseExceptionFilter {
+  /**
+   *
+   */
+  public catch(exception: unknown, host: ArgumentsHost): void {
+    const status_code = (exception as { status?: number }).status;
+
+    // don't report expected errors
+    if (status_code !== undefined && status_code >= 400 && status_code < 500) {
+      return super.catch(exception, host);
+    }
+
+    captureException(exception);
+    return super.catch(exception, host);
+  }
+}
+
+/**
+ * Set up a nest service that provides error handling and performance tracing.
+ */
+export class SentryIntegrationService implements OnModuleInit {
+  /**
+   * Called when the SentryModuleIntegration gets initialized.
+   */
+  public onModuleInit(): void {
+    // Sadly, NestInstrumentation has no requestHook, so we need to add the attributes here
+    // We register this hook in this method, because if we register it in the integration `setup`,
+    // it would always run even for users that are not even using Nest.js
+    const client = getClient();
+    if (client) {
+      client.on('spanStart', span => {
+        addNestSpanAttributes(span);
+      });
+    }
+  }
+}
+
+/**
+ * Set up a root module that can be injected in nest applications.
+ */
+export class SentryIntegrationModule {
+  /**
+   * Called by the user to set the module as root module in a nest application.
+   */
+  public static forRoot(): DynamicModule {
+    return {
+      module: SentryIntegrationModule,
+      providers: [
+        SentryIntegrationService,
+        {
+          provide: APP_FILTER,
+          useClass: SentryGlobalFilter,
+        },
+        {
+          provide: APP_INTERCEPTOR,
+          useClass: SentryTracingInterceptor,
+        },
+      ],
+      exports: [SentryIntegrationService],
+    };
+  }
+}
+
+function addNestSpanAttributes(span: Span): void {
+  const attributes = spanToJSON(span).data || {};
+
+  // this is one of: app_creation, request_context, handler
+  const type = attributes['nestjs.type'];
+
+  // If this is already set, or we have no nest.js span, no need to process again...
+  if (attributes[SEMANTIC_ATTRIBUTE_SENTRY_OP] || !type) {
+    return;
+  }
+
+  span.setAttributes({
+    [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.http.otel.nestjs',
+    [SEMANTIC_ATTRIBUTE_SENTRY_OP]: `${type}.nestjs`,
+  });
+}
+
+Catch()(SentryGlobalFilter);
+Injectable()(SentryTracingInterceptor);
+Injectable()(SentryIntegrationService);
+Global()(SentryIntegrationModule);
+Module({
+  providers: [
+    SentryIntegrationService,
+    {
+      provide: APP_FILTER,
+      useClass: SentryGlobalFilter,
+    },
+    {
+      provide: APP_INTERCEPTOR,
+      useClass: SentryTracingInterceptor,
+    },
+  ],
+  exports: [SentryIntegrationService],
+})(SentryIntegrationModule);
```

Comment:
m: Can we move the decorators to the class definition? This is like a weird side-effect pattern.

Reply:
Moved the decorators directly below the class definitions + added a note on top explaining that we cannot use @ syntax so we need to resort to this alternative